### PR TITLE
Upgrade to .NET 6

### PR DESCRIPTION
**Pull request by csharpfritz**
Combining the findings of PRs #319, #327, #321

Adding C# 10 features, updating docs and save points

ToDo:
- [x] Migrate SRC to use .NET 6 File Scoped namespaces
- [x] Migrate SRC to use Global Usings
- [x] Migrate SRC to use Implicit Usings
- [x] Migrate SRC to use Minimal APIs
- [x] Migrate SRC to use JSON Code Generation
- [x] Migrate SRC to use EF Compiled Models
- [x] Rollout SRC changes to save-point/00
- [x] Rollout SRC changes to save-point/01
- [x] Rollout SRC changes to save-point/02
- [x] Rollout SRC changes to save-point/03
- [x] Rollout SRC changes to save-point/04
- [x] Rollout SRC changes to save-point/05
- [x] Rollout SRC changes to save-point/06
- [x] Rollout SRC changes to save-point/07
- [x] Rollout SRC changes to save-point/08
- [x] Rollout SRC changes to save-point/09
- [x] Update docs

### CODE REVIEW

**Comment by ylr-research** [comment]
This is great! I just closed #331 because I was missing updating the build props

File: Directory.Build.props [hunk 1]
```diff
@@ -1,8 +1,9 @@
 <Project>
     <PropertyGroup>
-        <AspNetCoreVersion>5.0.0</AspNetCoreVersion>
-        <BlazorVersion>5.0.0</BlazorVersion>
-        <EntityFrameworkVersion>5.0.0</EntityFrameworkVersion>
-        <SystemNetHttpJsonVersion>5.0.0</SystemNetHttpJsonVersion>
+        <AspNetCoreVersion>6.0.0</AspNetCoreVersion>
+        <BlazorVersion>6.0.0</BlazorVersion>
+        <EntityFrameworkVersion>6.0.0</EntityFrameworkVersion>
+				<TargetFrameworkVersion>net6.0</TargetFrameworkVersion>
```
Comment (danroth27) [inline Directory.Build.props:6]: Whitespace issue?

File: save-points/00-get-started/BlazingPizza.Client/BlazingPizza.Client.csproj [hunk 2]
```diff
@@ -1,7 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>$(TargetFrameworkVersion)</TargetFramework>
+		<ImplicitUsings>true</ImplicitUsings>
```
Comment (danroth27) [inline save-points/00-get-started/BlazingPizza.Client/BlazingPizza.Client.csproj:5]: Whitespace issue?

File: save-points/00-get-started/BlazingPizza.ComponentsLibrary/BlazingPizza.ComponentsLibrary.csproj [hunk 3]
```diff
@@ -1,7 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Razor">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>$(TargetFrameworkVersion)</TargetFramework>
+		<ImplicitUsings>true</ImplicitUsings>
```
Comment (danroth27) [inline save-points/00-get-started/BlazingPizza.ComponentsLibrary/BlazingPizza.ComponentsLibrary.csproj:5]: Whitespace issue?

File: save-points/00-get-started/BlazingPizza.ComponentsLibrary/LocalStorage.cs [hunk 4]
```diff
@@ -1,17 +1,15 @@
...
-    {
-        public static ValueTask<T> GetAsync<T>(IJSRuntime jsRuntime, string key)
-            => jsRuntime.InvokeAsync<T>("blazorLocalStorage.get", key);
+	public static ValueTask<T> GetAsync<T>(IJSRuntime jsRuntime, string key)
+			=> jsRuntime.InvokeAsync<T>("blazorLocalStorage.get", key);
 
-        public static ValueTask SetAsync(IJSRuntime jsRuntime, string key, object value)
-            => jsRuntime.InvokeVoidAsync("blazorLocalStorage.set", key, value);
+	public static ValueTask SetAsync(IJSRuntime jsRuntime, string key, object value)
+			=> jsRuntime.InvokeVoidAsync("blazorLocalStorage.set", key, value);
```
Comment (danroth27) [inline save-points/00-get-started/BlazingPizza.ComponentsLibrary/LocalStorage.cs:11]: Indentation issue?

File: save-points/00-get-started/BlazingPizza.ComponentsLibrary/LocalStorage.cs [hunk 5]
```diff
@@ -1,17 +1,15 @@
...
 
-        public static ValueTask DeleteAsync(IJSRuntime jsRuntime, string key)
-            => jsRuntime.InvokeVoidAsync("blazorLocalStorage.delete", key);
-    }
+	public static ValueTask DeleteAsync(IJSRuntime jsRuntime, string key)
+			=> jsRuntime.InvokeVoidAsync("blazorLocalStorage.delete", key);
```
Comment (danroth27) [inline save-points/00-get-started/BlazingPizza.ComponentsLibrary/LocalStorage.cs:14]: Indentation issue?

File: Directory.Build.props (hunk 1 above)
Comment (csharpfritz) [reply Directory.Build.props]: Converted back to spaces

File: save-points/00-get-started/BlazingPizza.ComponentsLibrary/BlazingPizza.ComponentsLibrary.csproj (hunk 3 above)
Comment (csharpfritz) [reply save-points/00-get-started/BlazingPizza.ComponentsLibrary/BlazingPizza.ComponentsLibrary.csproj]: Converted to spaces

File: save-points/00-get-started/BlazingPizza.ComponentsLibrary/LocalStorage.cs (hunk 4 above)
Comment (csharpfritz) [reply save-points/00-get-started/BlazingPizza.ComponentsLibrary/LocalStorage.cs]: Converted to spaces

File: save-points/00-get-started/BlazingPizza.ComponentsLibrary/LocalStorage.cs (hunk 5 above)
Comment (csharpfritz) [reply save-points/00-get-started/BlazingPizza.ComponentsLibrary/LocalStorage.cs]: Converted to spaces

**Comment by csharpfritz** [comment]
I'm generating an .editorConfig to add to this with a space indentation set at 4 spaces.  I will run `dotnet format` against all code in the workshop

**Pull request merged**